relay-server/src/metrics.rs

```diff
@@ -0,0 +1,200 @@
+use relay_common::metrics::{CounterMetric, HistogramMetric, SetMetric, TimerMetric};
+
+/// Set metrics used by Relay
+pub enum RelaySets {
+    /// Represents the number of active projects in the current slice of time
+    UniqueProjects,
+}
+
+impl SetMetric for RelaySets {
+    fn name(&self) -> &'static str {
+        match self {
+            RelaySets::UniqueProjects => "unique_projects",
+        }
+    }
+}
+
+/// Histogram metrics used by Relay.
+pub enum RelayHistograms {
+    /// The number of events in the queue as a percentage of the maximum number of events
+    /// that can be stored in the queue ( 0 ... the queue is empty, 1 ... the queue is full
+    /// and no additional events can be added).
+    EventQueueSizePct,
+    /// The number of events in the queue at the sampling moment.
+    EventQueueSize,
+    /// The event size as seen by Relay after it is extracted from a request
+    EventSizeBytesRaw,
+    /// The event size as seen by Relay after it is extracted Base64 decoded and unzipped.
```

Comment:
There are more encodings other than base64 or unzip. Let's keep it abstract and just mention that whatever content-encoding was applied has been decompressed.